### PR TITLE
Add filter by Execution Year to department courses page. FIST-61 #resolve

### PR DESCRIPTION
**Pull request by pasadinhas**

### CODE REVIEW

File: fenixedu-ist-cms-components/src/main/webapp/themes/fenixedu-units-theme/departmentCourses.html [hunk 1]
```diff
@@ -20,7 +20,18 @@
 -->
 {% extends "fenixedu-units-theme/base.html" %}
 
+
```
Comment (jcarvalho) [inline fenixedu-ist-cms-components/src/main/webapp/themes/fenixedu-units-theme/departmentCourses.html]: Please avoid submitting whitespace changes.

File: fenixedu-ist-cms-components/src/main/webapp/themes/fenixedu-units-theme/departmentCourses.html [hunk 2]
```diff
@@ -29,7 +40,7 @@
                     <h3>{{ competenceCourseGroupUnit.name.content }}</h3>
                     <ul class="list-unstyled">
                         {% for competenceCourse in competenceCourseGroupUnit.competenceCourses %}
-                            <li><a href="{{ competenceCourse.url }}">{{ competenceCourse.name.content }} ({{ competenceCourse.acronym }})</a></li>
+                        <li><a href="{{ competenceCourse.url }}">{{ competenceCourse.name.content }} ({{ competenceCourse.acronym }})</a></li>
```
Comment (jcarvalho) [inline fenixedu-ist-cms-components/src/main/webapp/themes/fenixedu-units-theme/departmentCourses.html]: More whitespace?

File: fenixedu-ist-cms-components/src/main/java/pt/ist/fenixedu/cmscomponents/domain/unit/components/UnitCourses.java [hunk 3]
```diff
@@ -104,4 +114,47 @@ public DepartmentUnit getPersonDepartmentUnit() {
         final Department department = employee == null ? null : employee.getCurrentDepartmentWorkingPlace();
         return department == null ? null : department.getDepartmentUnit();
     }
+
+    public List<ExecutionYear> getValidExecutionYearsForUnit(Unit unit) {
+        List<ScientificAreaUnit> scientificAreaUnits = unit.getSubUnits().stream()
+                .filter(Unit::isScientificAreaUnit)
+                .map(ScientificAreaUnit.class::cast)
+                .sorted(ScientificAreaUnit.COMPARATOR_BY_NAME_AND_ID)
+                .collect(toList());
```
Comment (jcarvalho) [inline fenixedu-ist-cms-components/src/main/java/pt/ist/fenixedu/cmscomponents/domain/unit/components/UnitCourses.java]: You are only creating this list so you can later iterate over it, why not just iterate over the stream directly?

Comment (pasadinhas) [inline fenixedu-ist-cms-components/src/main/java/pt/ist/fenixedu/cmscomponents/domain/unit/components/UnitCourses.java]: I've never really worked with streams, but will look into that.

Comment (jcarvalho) [inline fenixedu-ist-cms-components/src/main/java/pt/ist/fenixedu/cmscomponents/domain/unit/components/UnitCourses.java]: Nevermind my comment. You are iterating it within another iteration, using the stream directly would consume it.

**Pull request merged**